src/heading.js

```diff
@@ -27,46 +25,120 @@ export default class Heading extends Plugin {
 	 * @inheritDoc
 	 */
 	static get requires() {
-		return [ HeadingEngine ];
+		return [ Paragraph, HeadingEngine ];
 	}
 
 	/**
 	 * @inheritDoc
 	 */
 	init() {
 		const editor = this.editor;
-		const command = editor.commands.get( 'heading' );
-		const options = command.options;
-		const collection = new Collection();
-
-		// Add options to collection.
-		for ( const { id, label } of options ) {
-			collection.add( new Model( {
-				id, label
+		const dropdownItems = new Collection();
+		const options = this._getLocalizedOptions();
+		const commands = [];
+		let defaultOption;
+
+		for ( let option of options ) {
+			// Add the option to the collection.
+			dropdownItems.add( new Model( {
+				modelElement: option.modelElement,
+				label: option.title
```

Comment:
It may not look good but OTOH it makes sense. "Label" is a label in DOM, in the UI component. "Title" is any kind of title, it could be a label, but also aria attribute or plain text. This is where two of them meet.